lib/hooks/create/place.js

```diff
@@ -1,9 +1,39 @@
-const { get } = require('lodash');
+const { get, omit } = require('lodash');
+const { diff } = require('deep-diff');
 const { autoResolved, withInspectorate, withLicensing } = require('../../flow/status');
 
+const normalize = (path, key, lhs, rhs) => {
+  // ignore array order
+  if (key === 'holding' || key === 'suitability') {
+    Array.isArray(lhs) && lhs.sort();
+    Array.isArray(rhs) && rhs.sort();
+    return [lhs, rhs];
+  }
+
+  // handle different falsy values (null, empty string, undefined)
+  if (!lhs && !rhs) {
+    return [undefined, undefined];
+  }
+
+  return false;
+};
+
+const onlyRolesChanged = model => {
+  const currentModel = get(model, 'data.modelData');
+  const data = get(model, 'data.data');
+
+  const oldValues = omit(currentModel, ['id', 'deleted', 'createdAt', 'updatedAt', 'nacwoId', 'migratedId']);
+  const newValues = omit(data, ['roles']);
```

Comment:
We're _not_ omitting `roles` from the old value. Does that not mean that if the old value has roles then this will always detect changes.

Reply:
The old values come from the `task.data.modelData` and now that roles come from a join, it doesn't end up in the task.

Thinking about it, we might want to change this to look up the current values of the place from the db to compare against, although I don't think you can have multiple inflight tasks for a place so it shouldn't be possible to change the data elsewhere between this task being created and resolved?

Reply:
Unless the answer is that the current model doesn't have roles attached, in which case how does the task diff table render roles?

Reply:
Given there's a number of tickets coming to fix issues with the diff tables _not_ using the `modelData` to render the "before" state (and hence ending up weird once resolved) I think the `modelData` _should_ include the roles, and therefore should also be omitted here.

In short - omit `roles`